camel-core/src/main/java/org/apache/camel/processor/aggregate/AggregateProcessor.java

```diff
@@ -1448,6 +1448,12 @@ protected void doStop() throws Exception {
         if (recoverService != null) {
             camelContext.getExecutorServiceManager().shutdown(recoverService);
         }
+        
+        if (timeoutCheckerExecutorService != null) {
```

Comment:
@gsudharsan , I'm not sure, that it's related to the CAMEL-4037. I think the reason why the tests are fails, that shutdown of the executor timeoutCheckerExecutorService is controlled by  shutdownTimeoutCheckerExecutorService boolean flag but not null value of the instance itself.
@davsclaus , @oscerd what do you think about this?

Reply:
I guess you're right @dmvolod

Reply:
OK, I will change the code to check the boolean iso null check and run the tests again..